src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,136 @@
+package parser;
+
+import commands.Command;
+import exceptions.AliceException;
+import exceptions.MissingArgumentException;
+import storage.TaskList;
+import tasks.Task;
+import ui.Ui;
+
+/**
+ * Parses user input and triggers the corresponding actions,
+ * by activating TaskList and Ui.
+ */
+public class Parser {
+    private boolean isBye;
+    private final TaskList list;
+    private final Ui ui;
+
+    /**
+     * Initialises a Parser object with a TaskList and Ui.
+     *
+     * @param list the list for storing the tasks.
+     * @param ui the ui for printing output to the terminal.
+     */
+    public Parser(TaskList list, Ui ui) {
+        isBye = false;
+        this.list = list;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses user input, breaking down actions by commands.
+     *
+     * @param input the string to be parsed.
+     * @throws AliceException if command is invalid, or if task number is invalid, or if task is missing arguments.
+     */
+    public void parse(String input) throws AliceException {
+        String[] result = input.split(" ", 2);
+        Command command = Command.stringToCommand(result[0]);
+        Task addedTask = new Task("");
+
+        switch (command) {
+        case BYE:
+            isBye = true;
+            break;
+        case LIST:
+            ui.printListedTasks(list.listTasks(), list.getSize());
+            break;
+        case MARK:
+            int taskNumber;
+            try {
+                taskNumber = Integer.parseInt(result[1]);
+            } catch (NumberFormatException e) {
+                System.out.println(result[1] + " is not a number!");
+                System.out.println("------------------------------------------");
+                break;
+            }
+            Task markTask = list.markTask(taskNumber);
+            ui.printHandleTaskMessage(markTask, "mark");
+            break;
+        case UNMARK:
+            int taskNum;
+            try {
+                taskNum = Integer.parseInt(result[1]);
+            } catch (NumberFormatException e) {
+                System.out.println(result[1] + " is not a number!");
+                System.out.println("------------------------------------------");
+                break;
+            }
+            Task unmarkTask = list.unmarkTask(taskNum);
+            ui.printHandleTaskMessage(unmarkTask, "unmark");
+            break;
+        case DELETE:
+            int taskNo;
+            try {
+                taskNo = Integer.parseInt(result[1]);
+            } catch (NumberFormatException e) {
+                System.out.println(result[1] + " is not a number!");
+                System.out.println("------------------------------------------");
+                break;
+            }
```

Comment:
I like how you've implemented error handling for invalid task numbers with a try-catch block! It ensures that the program doesn't crash when a user inputs a non-numeric value. However, would it be more user-friendly to handle the error by providing a more detailed message or even prompting the user to re-enter a valid number? This could improve the user experience by guiding them on how to correct their input. What do you think about this approach?